frontend/src/employee-frontend/components/messages/MessageEditor.tsx

```diff
@@ -54,17 +76,62 @@ export default React.memo(function MessageEditor({
   receiverOptions,
   setSelectedReceivers,
   onSend,
-  hideEditor
+  onDiscard,
+  onClose,
+  draftContent
 }: Props) {
   const { i18n } = useTranslation()
+  const [newDraftRequested, setNewDraftRequested] = useState<boolean>(false)
+  const [draftId, setDraftId] = useState<UUID | undefined>(draftContent?.id)
+  const [draftSaveResult, setDraftSaveResult] = useState<Result<void>>()
   const [selectorChange, setSelectorChange] = useState<SelectorChange>()
   const [selectedReceiverOptions, setSelectedReceiverOptions] = useState<
     Option[]
   >(getSelected(selectedReceivers))
   const [selectedAccount, setSelectedAccount] = useState<Option>(
     defaultAccountSelection
   )
-  const [messageBody, setMessageBody] = useState<MessageBody>(emptyMessageBody)
+  const [messageBody, setMessageBody] = useState<MessageBody>(
+    draftContentToInitialState(draftContent)
+  )
+
+  const [contentTouched, setContentTouched] = useState(false)
+  const saveDraft = useRestApi(api.saveDraft, setDraftSaveResult)
+  const initDraft = useRestApi(api.initDraft, (res: Result<UUID>) => {
+    setNewDraftRequested(false)
+    if (res.isSuccess) {
+      setDraftId(res.value)
+    }
+  })
+
+  // initialize draft when requested
+  useEffect(() => {
+    if (newDraftRequested && !draftId) {
+      initDraft(selectedAccount.value)
+    }
+  }, [draftId, newDraftRequested, initDraft, selectedAccount.value])
+
+  // save draft on input change or ask for a new draft to be initialized
+  const debouncedMessageBody = useDebounce(messageBody, 2000)
```

Comment:
There's something weird going on with the UI at my end:

The only way for me to update the draft seems to be to change the `sender`. Editing any other fields doesn't seem to send any XHRs.

Reply:
Revamped the whole draft save logic. All fields use the same debounced save logic now.